aria/modeling/relationships.py

```diff
@@ -0,0 +1,402 @@
+    :param backref_kwargs: Extra kwargs for SQLAlchemy `backref`
+    :type backref_kwargs: {}
+    """
+
+    this_table = model_class.__tablename__
+    this_column_name = '{0}_id'.format(this_table)
+    this_foreign_key = '{0}.id'.format(this_table)
+
+    other_column_name = '{0}_id'.format(other_table)
+    other_foreign_key = '{0}.id'.format(other_table)
+
+    secondary_table = '{0}_{1}'.format(this_table, other_table)
+
+    if other_property is None:
+        other_property = formatting.pluralize(this_table)
+        if prefix is not None:
+            secondary_table = '{0}_{1}'.format(prefix, secondary_table)
+            other_property = '{0}_{1}'.format(prefix, other_property)
+
+    backref_kwargs = backref_kwargs or {}
+    backref_kwargs.setdefault('uselist', True)
+
+    relationship_kwargs = relationship_kwargs or {}
+    relationship_kwargs.setdefault('secondary', _get_secondary_table(
+        model_class.metadata,
+        secondary_table,
+        this_column_name,
+        other_column_name,
+        this_foreign_key,
+        other_foreign_key
+    ))
+
+    return _relationship(model_class, other_table, backref_kwargs, relationship_kwargs,
+                         other_property, dict_key=dict_key)
+
+
+def _relationship(model_class, other_table, backref_kwargs, relationship_kwargs, other_property,
+                  fk=None, other_fk=None, dict_key=None):
+    relationship_kwargs = relationship_kwargs or {}
+
+    if fk:
+        relationship_kwargs.setdefault('foreign_keys',
+                                       lambda: getattr(
+                                           _get_class_for_table(
+                                               model_class,
+                                               model_class.__tablename__),
+                                           fk))
+
+    elif other_fk:
+        relationship_kwargs.setdefault('foreign_keys',
+                                       lambda: getattr(
+                                           _get_class_for_table(
+                                               model_class,
+                                               other_table),
+                                           other_fk))
+
+    if dict_key:
+        relationship_kwargs.setdefault('collection_class',
+                                       attribute_mapped_collection(dict_key))
+
+    if other_property is False:
+        # No backref
+        return relationship(
+            lambda: _get_class_for_table(model_class, other_table),
+            **relationship_kwargs
+        )
+    else:
+        if other_property is None:
+            other_property = model_class.__tablename__
+        backref_kwargs = backref_kwargs or {}
+        return relationship(
+            lambda: _get_class_for_table(model_class, other_table),
+            backref=backref(other_property, **backref_kwargs),
+            **relationship_kwargs
+        )
+
+
+def _get_class_for_table(model_class, tablename):
```

Comment:
It receives both a class and a tablename, and returns a class. This was your function originally, I did not change it.